services/cards/src/actions/Cards.tsx

```diff
@@ -8,10 +8,51 @@ import {filtersCalculate} from './Filters';
 declare var require: any;
 const Tabletop = require('tabletop');
 
+interface ResultType {
+  cards: CardType[];
+  translations: TranslationsType;
+}
+
 export function downloadCards(): ((dispatch: Redux.Dispatch<any>) => void) {
   return (dispatch: Redux.Dispatch<any>) => {
     const store = getStore();
     dispatch(cardsLoading());
+
+    const keys = store.getState().filters.source.current.split(':')[1].split(',');
```

Comment:
Any chance this could be passed into the action?

Reply:
We could, but that would duplicate this logic into three places... which seems like it would cause its own set of problems

Reply:
I was thinking just pass the filters into `downloadCards`, not the keys themselves. The logic doesn't need to be duplicated that way.